lib/chef/cookbook_version.rb

```diff
@@ -532,7 +535,7 @@ def self.chef_server_rest
     end
 
     def chef_server_rest
-      self.class.chef_server_rest
+      @chef_server_rest ||= Chef::REST.new(Chef::Config[:chef_server_url])
```

Comment:
Actually I see its not using any global state in the class, its still pretty ugly because of the tight coupling, I'm just not sure what to do about it.